trackers/core/ksptracker/tracker.py

```diff
@@ -0,0 +1,277 @@
+                    confidence=dets.confidence[det_idx],
+                )
+                G.add_node(node)
+
+                # Connect to source if first frame
+                if frame_idx == 0:
+                    G.add_edge("source", node, weight=-node.confidence)
+
+                # Connect to sink if last frame
+                if frame_idx == len(all_detections) - 1:
+                    G.add_edge(node, "sink", weight=0)
+
+                # Connect to future frames within max_gap
+                future_range = range(
+                    frame_idx + 1,
+                    min(frame_idx + self.max_gap + 1, len(all_detections)),
+                )
+                for future_idx in future_range:
+                    future_dets = all_detections[future_idx]
+                    for future_det_idx in range(len(future_dets)):
+                        if future_dets.confidence[future_det_idx] < self.min_confidence:
+                            continue
+
+                        future_node = TrackNode(
+                            frame_id=future_idx,
+                            detection_id=future_det_idx,
+                            bbox=tuple(future_dets.xyxy[future_det_idx]),
+                            confidence=future_dets.confidence[future_det_idx],
+                        )
+
+                        if self._can_connect_nodes(node, future_node):
+                            weight = self._edge_cost(node, future_node)
+                            G.add_edge(node, future_node, weight=weight)
+
+        return G
+
+    def _update_detections_with_tracks(self, assignments: Dict) -> sv.Detections:
+        """Update detections with track IDs based on assignments.
+
+        Args:
+            assignments (Dict): Maps (frame_id, det_id) to track_id
+
+        Returns:
+            sv.Detections: Updated detections with tracker_ids assigned
+        """
+        all_detections = []
+        all_tracker_ids = []
+
+        for frame_idx, dets in enumerate(self.detection_buffer):
+            frame_tracker_ids = [-1] * len(dets)
+
+            for det_idx in range(len(dets)):
+                key = (frame_idx, det_idx)
+                if key in assignments:
+                    frame_tracker_ids[det_idx] = assignments[key]
+
+            all_detections.append(dets)
+            all_tracker_ids.extend(frame_tracker_ids)
+
+        final_detections = sv.Detections.merge(all_detections)
+        final_detections.tracker_id = np.array(all_tracker_ids)
+
+        return final_detections
+
+    def ksp(self, graph: nx.DiGraph) -> List[List[TrackNode]]:
+        """Find K-shortest paths in the graph.
+
+        Args:
+            graph (nx.DiGraph): Directed graph of detection nodes
+
+        Returns:
+            List[List[TrackNode]]: List of paths, each path is list of TrackNodes
+        """
+        paths: List[List[TrackNode]] = []
+        for path in nx.shortest_simple_paths(graph, "source", "sink", weight="weight"):
```

Comment:
`nx.shortest_simple_paths` generates all paths without repeated nodes between a source and target in order of increasing length, but that does not necessarily guarantee node-disjoint paths. This means that a single detection could be assigned to multiple tracks.